auslib/blobs/apprelease.py

```diff
@@ -929,6 +936,162 @@ def getReferencedReleases(self):
+                        matches = True
+                elif condition == "versions":
+                    for version in values:
+                        if matchVersion(version, updateQuery["version"]):
+                            matches = True
+                            break
+                condition_results.append(matches)
+
+            if all(condition_results):
+                # matched everything!
+                attrs.update(group["fields"])
+
+        for attr in self.interpolable_:
+            if attr in attrs:
+                attrs[attr] = attrs[attr].replace("%LOCALE%", updateQuery["locale"])
+                attrs[attr] = attrs[attr].replace("%locale%", updateQuery["locale"])
+
+        updateLine = "    <update"
+        for key in sorted(attrs.keys()):
+            value = attrs[key]
+            if isinstance(value, bool):
+                value = str(value).lower()
+            updateLine += ' {}="{}"'.format(key, value)
+        updateLine += ">"
+
+        return updateLine
+
+    def getApplicationVersion(self, platform, locale):
+        return self.getLocaleOrTopLevelParam(platform, locale, 'appVersion')
+
+    def validate(self, *args, **kwargs):
+        super(ReleaseBlobV9, self).validate(*args, **kwargs)
+
+        conflicts = []
+        conflicting_values = set()
+
+        for group1 in self["updateLine"]:
+            for group2 in self["updateLine"]:
+                # Skip over groups that are identical - they can't conflict
+                if group1 == group2:
+                    continue
+                # If there are no overlapping fields between the groups, there can't be a conflict
+                field_dupes = set(group1["fields"].keys()).intersection(group2["fields"].keys())
+                if len(field_dupes) == 0:
+                    continue
+                # If there are overlapping fields, we need to see if what they apply to overlap in any way
+                condition_results = []
+                for cond in ("channels", "locales", "versions"):
+                    matches = False
+                    # If a condition is not specified for a group, it always matches anything
+                    if cond not in group1["for"] or cond not in group2["for"]:
+                        matches = True
+                    else:
+                        if cond == "channels":
+                            for (value1, value2) in itertools.product(group1["for"][cond], group2["for"][cond]):
+                                # Exact match of concrete channel or two globs
+                                if value1 == value2:
+                                    matches = True
+                                    break
+                                # Logical match in either direction, which takes into account
+                                # globbing and fallback channels
+                                elif matchChannel(value1, value2.rstrip("*"), getFallbackChannel(value2.rstrip("*"))):
```

Comment:
This would fail in the case of release-cdntest vs release* ?

Reply:
I think that one works:
```
>>> matchChannel("release*", "release-cck-", "release")
True
```

I'll see about fleshing out the tests more, though.